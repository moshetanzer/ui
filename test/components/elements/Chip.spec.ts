import { describe, it, expect } from 'vitest'
import { UChip } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Chip', () => {
  it.each([
    ['basic case', {}],
    ['with custom size', { props: { size: 'lg' } }],
    ['with custom color', { props: { color: 'red' } }],
    ['with custom position', { props: { position: 'top-left' } }],
    ['with inset chip', { props: { inset: true } }],
    ['with hidden chip', { props: { show: false } }],
    ['with content slot', { slots: { content: () => 'Content slot' } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UChip.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UChip)
    expect(html).toMatchSnapshot()
  })
})
