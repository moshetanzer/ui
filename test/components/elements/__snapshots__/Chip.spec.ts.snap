// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Chip > renders basic case correctly 1`] = `"<div class="relative inline-flex items-center justify-center flex-shrink-0"><span class="absolute rounded-full ring-1 ring-white dark:ring-gray-900 flex items-center justify-center text-white dark:text-gray-900 font-medium whitespace-nowrap h-2 min-w-[0.5rem] text-[7px] p-0.5 top-0 right-0 -translate-y-1/2 translate-x-1/2 transform bg-primary-500 dark:bg-primary-400"></span></div>"`;

exports[`Chip > renders with content slot correctly 1`] = `"<div class="relative inline-flex items-center justify-center flex-shrink-0"><span class="absolute rounded-full ring-1 ring-white dark:ring-gray-900 flex items-center justify-center text-white dark:text-gray-900 font-medium whitespace-nowrap h-2 min-w-[0.5rem] text-[7px] p-0.5 top-0 right-0 -translate-y-1/2 translate-x-1/2 transform bg-primary-500 dark:bg-primary-400">Content slot</span></div>"`;

exports[`Chip > renders with custom color correctly 1`] = `"<div class="relative inline-flex items-center justify-center flex-shrink-0"><span class="absolute rounded-full ring-1 ring-white dark:ring-gray-900 flex items-center justify-center text-white dark:text-gray-900 font-medium whitespace-nowrap h-2 min-w-[0.5rem] text-[7px] p-0.5 top-0 right-0 -translate-y-1/2 translate-x-1/2 transform bg-red-500 dark:bg-red-400"></span></div>"`;

exports[`Chip > renders with custom position correctly 1`] = `"<div class="relative inline-flex items-center justify-center flex-shrink-0"><span class="absolute rounded-full ring-1 ring-white dark:ring-gray-900 flex items-center justify-center text-white dark:text-gray-900 font-medium whitespace-nowrap h-2 min-w-[0.5rem] text-[7px] p-0.5 top-0 left-0 -translate-y-1/2 -translate-x-1/2 transform bg-primary-500 dark:bg-primary-400"></span></div>"`;

exports[`Chip > renders with custom size correctly 1`] = `"<div class="relative inline-flex items-center justify-center flex-shrink-0"><span class="absolute rounded-full ring-1 ring-white dark:ring-gray-900 flex items-center justify-center text-white dark:text-gray-900 font-medium whitespace-nowrap h-3 min-w-[0.75rem] text-[10px] p-0.5 top-0 right-0 -translate-y-1/2 translate-x-1/2 transform bg-primary-500 dark:bg-primary-400"></span></div>"`;

exports[`Chip > renders with hidden chip correctly 1`] = `
"<div class="relative inline-flex items-center justify-center flex-shrink-0">
  <!--v-if-->
</div>"
`;

exports[`Chip > renders with inset chip correctly 1`] = `"<div class="relative inline-flex items-center justify-center flex-shrink-0"><span class="absolute rounded-full ring-1 ring-white dark:ring-gray-900 flex items-center justify-center text-white dark:text-gray-900 font-medium whitespace-nowrap h-2 min-w-[0.5rem] text-[7px] p-0.5 top-0 right-0 bg-primary-500 dark:bg-primary-400"></span></div>"`;
